src/pages/Staking/VestsList/DelegateVest/DelegateVest.container.tsx

```diff
@@ -1,28 +1,91 @@
+import { useCallback, useEffect, useState } from 'react';
 import { useSelector } from 'react-redux';
+import { SubmitStatusDialog } from '../../../../components/TxDialog/TxDialog.component';
+import { Vesting } from '../../../../contracts/types';
+import { useContractCall } from '../../../../hooks/useContractCall';
+import {
+  accountSelector,
+  providerSelector,
+} from '../../../../store/app/app.selectors';
 
-import { selectedVestSelector } from '../../../../store/staking/staking.selectors';
+import {
+  combinedVotingPowerSelector,
+  selectedVestSelector,
+} from '../../../../store/staking/staking.selectors';
+import { getVesting } from '../../../../store/staking/staking.utils';
+import { DelegateFeeEstimator } from '../../Staking.types';
 
 import { DelegateVestComponent } from './DelegateVest.component';
+import { DelegateValues } from '../../DelegateFeeEstimator/DelegateVest.fields';
 import { DelegateVestContainerProps } from './DelegateVest.types';
 
-const mockVotingPower = '15.0000';
-
 export const DelegateVestContainer = ({
   open,
   onClose,
 }: DelegateVestContainerProps) => {
+  const account = useSelector(accountSelector);
   const selectedVestData = useSelector(selectedVestSelector);
+  const combinedVotingPower = useSelector(combinedVotingPowerSelector);
+  const provider = useSelector(providerSelector);
+  const [vesting, setVesting] = useState<Vesting | undefined>();
+
+  useEffect(() => {
+    if (!provider || !selectedVestData) {
+      throw new Error('missing data');
+    }
+    const vested = getVesting(selectedVestData.address, provider);
```

Comment:
I this case when you have the vestingAddress in the store, you could create a selector that will return the VestingContract connected to proper address